### PR TITLE
Upgrade to Sylius 1.9 and Doctrine migration 3

### PR DESCRIPTION
**Pull request by Roshyo**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,9 +10,10 @@
     ],
     "license": "MIT",
     "require": {
-        "php": "^7.3",
+        "php": "^7.4 || ^8.0",
+        "friendsofsymfony/oauth-server-bundle": ">2.0.0-alpha.0 ^2.0@dev",
         "setono/doctrine-orm-batcher-bundle": "^0.3.1",
-        "sylius/sylius": "^1.3",
+        "sylius/sylius": "^1.9",
```
Comment (loevgaard) [inline composer.json:16]:
```suggestion
        "sylius/sylius": "~1.9.0",
```

File: tests/Application/config/packages/_sylius.yaml [hunk 2]
```diff
@@ -6,15 +6,11 @@ imports:
...
+
 parameters:
     sylius_core.public_dir: '%kernel.project_dir%/public'
 
-sylius_product:
-    resources:
-        product:
-            classes:
-                model: Tests\Setono\SyliusCalloutPlugin\Entity\Product
-
```
Comment (loevgaard) [inline tests/Application/config/packages/_sylius.yaml:17]: Why is this removed?

File: tests/Application/src/Resources/config/doctrine/Product.orm.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
                                       http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd"
 >
 
-    <entity name="Tests\Setono\SyliusCalloutPlugin\Entity\Product" table="sylius_product">
+    <mapped-superclass name="Tests\Setono\SyliusCalloutPlugin\Entity\Product" table="sylius_product">
```
Comment (loevgaard) [inline tests/Application/src/Resources/config/doctrine/Product.orm.xml:9]: I guess there's no reason for this change and I do believe it's more correct to use `entity` when you're in an app context

File: composer.json (hunk 1 above)
Comment (Roshyo) [reply composer.json:16]: Why would you restrict it ? I believe we can keep 1.9 and more, don't you think ?

Comment (loevgaard) [reply composer.json:16]: I like the constraint for Sylius because then when we return to this plugin one year from now we know which version worked. Sylius often times breaks between minor releases because of all sorts of config changes. This way the test application will always work

File: composer.json [hunk 4]
```diff
@@ -10,9 +10,10 @@
     ],
     "license": "MIT",
     "require": {
-        "php": "^7.3",
+        "php": "^7.4 || ^8.0",
```
Comment (loevgaard) [inline composer.json:13]:
```suggestion
        "php": ">=7.4",
```
